robot_skills/src/robot_skills/robot.py

```diff
@@ -129,34 +161,136 @@ def tf_transform_pose(self, ps, frame):
+        :param desired_objects: Collection of objects that the arm may have. None
+                means there are no desired objects.
+
+        :return: An Arm of the robot with the requested properties, or None.
+                Note that the arm will never do more than you requested.
+        """
+        discarded_reasons = [] # Reasons why arms are discarded.
+
+        for arm_name, arm in self.iteritems():
+            matching_arm = arm
+            matching_grippers = None
+            matching_goals = None
+            matching_trajectories = None
+            uses_objects = None
+
+            if not arm.operational:
+                discarded_reasons.append((arm_name, "not operational"))
+                continue
+
+            # Name
+            if required_arm_name is not None and arm_name != required_arm_name:
+                discarded_reasons.append((arm_name, "required arm name failed"))
+                continue
+
+            # Grippers
+            matching_grippers = set()
+            if required_gripper_types is not None:
+                matches = [self.collect_gripper_types(req_type) for req_type in required_gripper_types]
+                all_matched = all(match_list for match_list in matches)
+                if not all_matched:
+                    discarded_reasons.append((arm_name, "required gripper type failed"))
+                    continue
+                matching_grippers.update(matches)
+
+            if desired_gripper_types is not None:
+                matching_grippers.update(self.collect_gripper_types(des_type) for des_type in desired_gripper_types)
+
+            # Goals
+            matching_goals = _collect_needs_desires(required_goals, desired_goals, arm.has_joint_goal)
+            if matching_goals is None:
+                discarded_reasons.append((arm_name, "required goals failed"))
+                continue
+
+            # Trajectories
+            matching_trajectories = _collect_needs_desires(required_trajectories, desired_trajectories,
+                                                           arm.has_joint_trajectory)
+            if matching_trajectories is None:
+                discarded_reasons.append((arm_name, "required trajectories failed"))
+                continue
+
+            # Objects
+            if not self._check_required_obj(required_objects):
+                discarded_reasons.append((arm_name, "required objects failed"))
+                continue
+            # Desired objects not checked currently.
+
+            # Unlike the other properties, it is likely not exactly known what objects will be used.
+            # For this reason, it seems better not to limit objects to the set specified in required
+            # _objects and desired_objects.
+            uses_objects = (required_objects is not None or desired_objects is not None)
+
+
+            # Success!
+            return arms.PublicArm(matching_arm, matching_gripper, uses_objects,
```

Comment:
De `matching_gripper` wordt niet gekozen uit de mathing grippers?!

Reply:
Code did miss a default gripper (fixed now), but no, in general an arm can have several grippers.